server.go

```diff
@@ -602,13 +752,18 @@ func (s *Server) Stop() {
 	s.lis = nil
 	cs := s.conns
 	s.conns = nil
+	for c := range s.hconns {
+		go c.Close()
```

Comment:
Doesn't doing this asynchronously violate least surprise for this function? It may return before all connections are closed.